doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+Each pipeline stage has some associated state.
+
+* IF stage full or empty
+* WB stage full or empty
+* ID stage full or empty -  effectively covered by instruction categories (Category None meaning the stage is empty).
+* Controller (within ID stage) state machine states
+
+  * Possible transitions between these states.
+    Those marked with a '*' are of particular interest and should be crossed with instruction categories and other coverpoints as appropriate to fully explore the transitions.
+
+    * ``RESET`` -> ``BOOT_SET``
+    * ``BOOT_SET`` -> ``FIRST_FETCH``
+    * ``FIRST_FETCH`` -> ``DECODE``
+    * ``FIRST_FETCH`` -> ``IRQ_TAKEN``
+    * ``FIRST_FETCH`` -> ``DBG_TAKEN_IF``
+    * ``DECODE`` -> ``FLUSH`` *
+    * ``DECODE`` -> ``DBG_TAKEN_IF`` *
+    * ``DECODE`` -> ``IRQ_TAKEN`` *
+    * ``IRQ_TAKEN`` -> ``DECODE``
+    * ``DBG_TAKEN_IF`` -> ``DECODE``
+    * ``DBG_TAKEN_ID`` -> ``DECODE``
+    * ``FLUSH`` -> ``DECODE`` *
+    * ``FLUSH`` -> ``DBG_TAKEN_ID``
+    * ``FLUSH`` -> ``WAIT_SLEEP``
+    * ``FLUSH`` -> ``IRQ_TAKEN`` *
+    * ``FLUSH`` -> ``DBG_TAKEN_IF`` *
+    * ``WAIT_SLEEP`` -> ``SLEEP``
+    * ``SLEEP`` -> ``FIRST_FETCH``
+
+Exceptions/Interrupts/Debug
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Exceptions, interrupts and debug entry can all cause control flow changes combined with CSR writes and privilege level changes and work quite similarly within the controller but not identically.
+Furthermore they can all occur together and must be appropriately prioritised (consider a instruction with hardware trigger point matching it, that causes some exception and an interrupt is raised the cycle it enters the ID/EX stage)
+
+* Exception from instruction fetch error (covered by FetchError instruction category).
+* Exception from instruction PMP violation.
+* Exception from illegal instruction (covered by Illegal instruction categories).
+* Exception from memory fetch error.
+* Exception from memory access PMP violation.
+* Unaligned access cases (both accesses saw error, first or second only saw error or neither saw error) for both kinds of memory exceptions.
+* Interrupt raised/taken.
+
+  * Interrupt raised/taken for each available interrupt line (for wider cross coverage which interrupt is raised/taken is not relevant other than NMI vs non NMI).
+  * Interrupt raised/taken first cycle of instruction or not.
+
+* External debug request.
+* Instruction executed when debug single step enabled.
+* Instruction matches hardware trigger point.
+* Ibex operating in debug mode.
+
+PMP
+^^^
+* Each region configured with different matching modes.
+
+  * Off
+  * TOR
+  * NA4
+  * NAPOT
+
+* Each region configured with all possible permissions including locked/unlocked.
+
+  * Different permissions with MML enabled and disabled, separate cover points for R/W/X/L values with and without MML.
+
+* Access fail & pass.
+
+  * All combinations of unaligned access split across a boundary, both halves pass, neither pass, one does & one doesn’t.
```

Comment:
Maybe the last item should be expanded to "just the first passes, just the second passes"?